src/authenticationservices.cs

```diff
@@ -0,0 +1,195 @@
+//
+// AuthenticationServices bindings
+//
+// Authors:
+//	Sebastien Pouliot  <sebastien.pouliot@microsoft.com>
+//
+// Copyright 2018 Microsoft Inc. All rights reserved.
+//
+
+using System;
+using Foundation;
+using ObjCRuntime;
+using UIKit;
+
+namespace AuthenticationServices {
+
+	[iOS (12,0)]
+	[Native]
+	[ErrorDomain ("ASCredentialIdentityStoreErrorDomain")]
+	public enum ASCredentialIdentityStoreErrorCode : long {
+		InternalError = 0,
+		StoreDisabled = 1,
+		StoreBusy = 2,
+	}
+
+	[iOS (12,0)]
+	[Native]
+	[ErrorDomain ("ASExtensionErrorDomain")]
+	public enum ASExtensionErrorCode : long {
+		Failed = 0,
+		UserCanceled = 1,
+		UserInteractionRequired = 100,
+		CredentialIdentityNotFound = 101,
+	}
+
+	[iOS (12,0)]
+	[Native]
+	public enum ASCredentialServiceIdentifierType : long {
+		Domain,
+		Url,
+	}
+
+	[iOS (12,0)]
+	[Native]
+	[ErrorDomain ("ASWebAuthenticationSessionErrorDomain")]
+	public enum ASWebAuthenticationSessionErrorCode : long {
+		CanceledLogin = 1,
+	}
+
+	[iOS (12,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASCredentialIdentityStore {
+		[Static]
+		[Export ("sharedStore")]
+		ASCredentialIdentityStore SharedStore { get; }
+
+		[Async]
+		[Export ("getCredentialIdentityStoreStateWithCompletion:")]
+		void GetCredentialIdentityStoreState (Action<ASCredentialIdentityStoreState> completion);
+
+		[Async]
+		[Export ("saveCredentialIdentities:completion:")]
+		void SaveCredentialIdentities (ASPasswordCredentialIdentity[] credentialIdentities, [NullAllowed] Action<bool, NSError> completion);
```

Comment:
Minor:

I am not sure I like `Action<bool, NSError>` here, what does that `bool` means? also the same delegate can be shared below I guess